.github/workflows/test_on_push.yml

```
@@ -87,3 +90,36 @@ jobs:
 
       - name: Check if the documentation can be built
         run: python -m nox -s docs
+
+  generated_project_tests:
+      needs: [style, template_test]
```

Since `template_test` needs `style`, you can exclude `style` here and the DAG can then look like this:

```mermaid
flowchart TD;
    A[style] --> B[template_test];
    B --> C[generated_project_tests];
```
instead of this:

```mermaid
flowchart TD;
    A[style] --> C[template_test];
    B[template_test] --> C[generated_project_tests];
```

```suggestion
      needs: [template_test]

```